Checkout/Source/Tokenisation/CheckoutAPIService.swift

```diff
@@ -187,6 +182,47 @@ final public class CheckoutAPIService: CheckoutAPIProtocol {
     }
   }
 
+  private func callRiskSDK(tokenDetails: TokenDetails,
+                           completion: @escaping () -> Void) {
+
+    /* Risk SDK calls can be finalised in 3 different ways
+     1. When Risk SDK's configure(...) function completed successfully and publishData(...) completed successfully or not
+     2. When Risk SDK's configure(...) function completed with failure
+     3. When Risk SDK's configure(...) or publishData(...) functions hang and don't call their completion blocks.
+        In this case, we wait for `self.timeoutInterval` amount of time and call the completion block anyway.
+
+     All these operations are done synchronously to avoid the completion closure getting called multiple times.
+    */
+
+    let finaliseRiskSDKCalls = {
+      self.taskCompletionQueue.sync {
+        if !self.isTaskCompleted {
+          self.isTaskCompleted = true
+          completion()
+        }
+      }
+    }
+
+    DispatchQueue.global(qos: .userInitiated).asyncAfter(deadline: .now() + timeoutInterval) {
+      finaliseRiskSDKCalls()
+    }
+
+    self.riskSDK.configure { [weak self] configurationResult in
+      guard let self else { return }
+      switch configurationResult {
```

Comment:
The `isTaskCompleted` variable is accessed from multiple threads (`main` thread and `taskCompletionQueue`). To ensure thread safety, you should protect its access with a lock or use an atomic property.

```suggestion
        self.taskCompletionQueue.sync {
      switch configurationResult {
```

Reply:
`finaliseRiskSDKCalls` closure has that check already